elastalert/ruletypes/maas_aggregation_rule.py

```diff
@@ -0,0 +1,153 @@
+from abc import ABC
+
+from elastalert.clients.mlflow_client import MaasClientMapper
+from elastalert.queries.elasticsearch_query import ElasticsearchMaasAggregationQuery
+from elastalert.queries.query_factory import QueryFactory
+from elastalert.ruletypes.base_aggregation_rule import BaseAggregationRule
+from elastalert.utils.maas_filter import FilterMapper
+
+
+class MaasAggregationRule(BaseAggregationRule, ABC):
```

Comment:
Why is this class a child of ABC?

Reply:
Removed. An automatic fix from pycharm lead to this error. Pycharm shows the methods add_data, add_count_data, add_terms_data and garbage_collect as abstract methods and suggests to register this class as abstract or to implement the methods. The mentioned methods are defined as @deprecated and raise a NotImplementationError.

We could remove the mentioned methods as they are not necessary (to my understanding). Or let them stay to not break the api. Nevertheless add_* functions must be registered manually as a callback in the rules (init_query_factory). 

In contrast to the add* methods, the garbage_collect method is defined as an abstract method, but a method body = pass. I suggest either removing the @abstractmethod decleration or implementing the method by hand in all subtypes of rule, where the method is not overwritten and the abstract method is taken.